lib/index.js

```diff
@@ -263,16 +312,72 @@ api.get = async ({actor, account, email, type} = {}) => {
   } else {
     query['account.email'] = email;
   }
+
   const record = await database.collections.account.findOne(
     query, {projection: {_id: 0, [typeKey]: 1}});
+
   if(!record) {
     throw new BedrockError('Authentication token not found.', 'NotFoundError', {
       httpStatusCode: 404,
       public: true
     });
   }
 
-  return record.meta['bedrock-authn-token'].tokens[type];
+  const result = record.meta['bedrock-authn-token'].tokens[type];
+  if(type === 'nonce') {
+    if(tokenId) {
+      return result.filter(token => token.id === tokenId)[0];
+    } else {
+      return result[0];
+    }
```

Comment:
- Should it return 404 if tokenId isn't found?  I think it would just return an empty array here?
- Is returning the frist item ok when tokenId not specified?
- Does the 'expires' field need to be taken into account in `get` and `getAll`?  Or is just handling that in `set` ok?

Reply:
@davidlehn Good points. You're right, it makes sense to return an error instead, I don't know what I was thinking lol. I can change that. 
Regarding  taking `expires` field into account in `get` and `getAll`, is there a use case for it?

Reply:
It depends what the callers are expecting.  If the expectation is they will inspect the expires field as needed, then no problem.  If the expectation was to get unexpired tokens, then the logic would have to be here.  I'm not sure what all the use cases are for this API to say what's needed.  Should probably make this code work in whatever way is needed now and add a flag later to change behavior if needed.  @dlongley?

Reply:
I think we should drop any expired tokens from the result -- as we are probably likely to make them auto-expire with a mongodb index in the future anyway. Is that approach a problem for any reason?

Reply:
Just performance issues of when these checks and removals happen.  Right now it's doing actual removal in `set`.  Should `get[All]` do removal too or just filter out expired tokens (and 404 if none are left)?

Reply:
@dlongley @davidlehn I haven't updated this yet, I'll make the changes when a decision has been made on this issue.

Reply:
I don't think `get` should do removals. It should just do 404 when a specific token isn't found. Similarly, `getAll` should not do removals, it should just return an empty array if all are expired.